Add genai docs (#19750)
microsoft/onnxruntime: docs/genai/tutorials/phi2-python.md

---
title: Python phi-2 tutorial
description: Learn how to write a language generation application with ONNX Runtime GenAI in Python using the phi-2 model
has_children: false
parent: Tutorials
grand_parent: Generative AI (Preview)
nav_order: 1
---

# Language generation in Python with phi-2

## Setup and installation

Install the ONNX Runtime GenAI Python package using the [installation instructions](../howto/install.md).

## Build phi-2 ONNX model

> The onnxruntime-genai package contains a model builder that generates the phi-2 ONNX model using the weights and config on Huggingface. The tools also allows you to load locally stored weights, or convert from GGUF format. For more details, see [how to build models](../howto/build-models.md)

Reviewer: ```suggestion
The onnxruntime-genai package contains a model builder that generates the phi-2 ONNX model using the weights and config on Huggingface. The tools also allows you to download the weights from Hugging Face, load locally stored weights, or convert from GGUF format. For more details, see [how to build models](../howto/build-models.md)
```

> If using the `-m` option shown here, which downloads from HuggingFace, you will need to login into HuggingFace.

Reviewer: ```suggestion
If using the `-m` option shown here, you will need to login into Hugging Face.
```

```bash
pip install huggingface-hub`
huggingface-cli --login
```

You can build the model in different precisions. This command uses int4 as it produces the smallest model and can run on a CPU.

```bash
python -m onnxruntime_genai.models.builder -m microsoft/phi-2 -e cpu -p int4 -o ./example-models/phi2-int4-cpu
```
You can replace the name of the output folder specified with the `-o` option with a folder of your choice.

After you run the script, you will see a series of files generated in this folder. They include the HuggingFace configs for your reference, as well as the following generated files used by ONNX Runtime GenAI.

`model.onnx`: the phi-2 ONNX model

> `model.onnx.data`: the phi-2 ONNX model weights

Reviewer: These don't appear as a list in the staged docs

Author: Fixed
